[2025-06-02T02:03:09.640+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-06-02T02:03:09.660+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: log_monitoring_dag.start_spark_consumer manual__2025-06-02T01:38:51.955055+00:00 [queued]>
[2025-06-02T02:03:09.672+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: log_monitoring_dag.start_spark_consumer manual__2025-06-02T01:38:51.955055+00:00 [queued]>
[2025-06-02T02:03:09.673+0000] {taskinstance.py:2865} INFO - Starting attempt 4 of 4
[2025-06-02T02:03:09.692+0000] {taskinstance.py:2888} INFO - Executing <Task(BashOperator): start_spark_consumer> on 2025-06-02 01:38:51.955055+00:00
[2025-06-02T02:03:09.699+0000] {logging_mixin.py:190} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=859) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-06-02T02:03:09.701+0000] {standard_task_runner.py:72} INFO - Started process 860 to run task
[2025-06-02T02:03:09.699+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'log_monitoring_dag', 'start_spark_consumer', 'manual__2025-06-02T01:38:51.955055+00:00', '--job-id', '76', '--raw', '--subdir', 'DAGS_FOLDER/log_monitoring_dag.py', '--cfg-path', '/tmp/tmp1vpuufr8']
[2025-06-02T02:03:09.702+0000] {standard_task_runner.py:105} INFO - Job 76: Subtask start_spark_consumer
[2025-06-02T02:03:09.720+0000] {logging_mixin.py:190} WARNING - /home/***/.local/lib/python3.12/site-packages/***/settings.py:209 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
[2025-06-02T02:03:09.760+0000] {task_command.py:467} INFO - Running <TaskInstance: log_monitoring_dag.start_spark_consumer manual__2025-06-02T01:38:51.955055+00:00 [running]> on host 95af1ae74a08
[2025-06-02T02:03:09.854+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='log_monitoring_dag' AIRFLOW_CTX_TASK_ID='start_spark_consumer' AIRFLOW_CTX_EXECUTION_DATE='2025-06-02T01:38:51.955055+00:00' AIRFLOW_CTX_TRY_NUMBER='4' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-06-02T01:38:51.955055+00:00'
[2025-06-02T02:03:09.855+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-06-02T02:03:09.877+0000] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2025-06-02T02:03:09.879+0000] {subprocess.py:75} INFO - Running command: ['/usr/bin/bash', '-c', '\n            docker run --rm             --network realtime_log_monitoring_default             -v /opt/***/app:/app             bitnami/spark:3.3.0             spark-submit --packages org.apache.spark:spark-sql-kafka-0-10_2.12:3.2.0 /app/log_consumer.py\n        ']
[2025-06-02T02:03:09.892+0000] {subprocess.py:86} INFO - Output:
[2025-06-02T02:03:13.433+0000] {subprocess.py:93} INFO - [38;5;6m [38;5;5m02:03:13.41 [0m
[2025-06-02T02:03:13.435+0000] {subprocess.py:93} INFO - [38;5;6m [38;5;5m02:03:13.43 [0m[1mWelcome to the Bitnami spark container[0m
[2025-06-02T02:03:13.437+0000] {subprocess.py:93} INFO - [38;5;6m [38;5;5m02:03:13.43 [0mSubscribe to project updates by watching [1mhttps://github.com/bitnami/containers[0m
[2025-06-02T02:03:13.452+0000] {subprocess.py:93} INFO - [38;5;6m [38;5;5m02:03:13.45 [0mSubmit issues and feature requests at [1mhttps://github.com/bitnami/containers/issues[0m
[2025-06-02T02:03:13.453+0000] {subprocess.py:93} INFO - [38;5;6m [38;5;5m02:03:13.45 [0m
[2025-06-02T02:03:13.463+0000] {subprocess.py:93} INFO - 
[2025-06-02T02:03:15.236+0000] {subprocess.py:93} INFO - :: loading settings :: url = jar:file:/opt/bitnami/spark/jars/ivy-2.5.0.jar!/org/apache/ivy/core/settings/ivysettings.xml
[2025-06-02T02:03:15.287+0000] {subprocess.py:93} INFO - Ivy Default Cache set to: /opt/bitnami/spark/.ivy2/cache
[2025-06-02T02:03:15.288+0000] {subprocess.py:93} INFO - The jars for the packages stored in: /opt/bitnami/spark/.ivy2/jars
[2025-06-02T02:03:15.291+0000] {subprocess.py:93} INFO - org.apache.spark#spark-sql-kafka-0-10_2.12 added as a dependency
[2025-06-02T02:03:15.292+0000] {subprocess.py:93} INFO - :: resolving dependencies :: org.apache.spark#spark-submit-parent-ed84d9e7-fd06-49e3-bc87-fc4719c53d8a;1.0
[2025-06-02T02:03:15.292+0000] {subprocess.py:93} INFO - 	confs: [default]
[2025-06-02T02:03:19.702+0000] {subprocess.py:93} INFO - 	found org.apache.spark#spark-sql-kafka-0-10_2.12;3.2.0 in central
[2025-06-02T02:03:20.195+0000] {subprocess.py:93} INFO - 	found org.apache.spark#spark-token-provider-kafka-0-10_2.12;3.2.0 in central
[2025-06-02T02:03:20.632+0000] {subprocess.py:93} INFO - 	found org.apache.kafka#kafka-clients;2.8.0 in central
[2025-06-02T02:03:20.917+0000] {subprocess.py:93} INFO - 	found org.lz4#lz4-java;1.7.1 in central
[2025-06-02T02:03:21.222+0000] {subprocess.py:93} INFO - 	found org.xerial.snappy#snappy-java;1.1.8.4 in central
[2025-06-02T02:03:22.739+0000] {subprocess.py:93} INFO - 	found org.slf4j#slf4j-api;1.7.30 in central
[2025-06-02T02:03:25.432+0000] {subprocess.py:93} INFO - 	found org.apache.hadoop#hadoop-client-runtime;3.3.1 in central
[2025-06-02T02:03:26.853+0000] {subprocess.py:93} INFO - 	found org.spark-project.spark#unused;1.0.0 in central
[2025-06-02T02:03:27.177+0000] {subprocess.py:93} INFO - 	found org.apache.hadoop#hadoop-client-api;3.3.1 in central
[2025-06-02T02:03:29.918+0000] {subprocess.py:93} INFO - 	found org.apache.htrace#htrace-core4;4.1.0-incubating in central
[2025-06-02T02:03:32.659+0000] {subprocess.py:93} INFO - 	found commons-logging#commons-logging;1.1.3 in central
[2025-06-02T02:03:33.042+0000] {subprocess.py:93} INFO - 	found com.google.code.findbugs#jsr305;3.0.0 in central
[2025-06-02T02:03:35.253+0000] {subprocess.py:93} INFO - 	found org.apache.commons#commons-pool2;2.6.2 in central
[2025-06-02T02:03:35.397+0000] {subprocess.py:93} INFO - downloading https://repo1.maven.org/maven2/org/apache/spark/spark-sql-kafka-0-10_2.12/3.2.0/spark-sql-kafka-0-10_2.12-3.2.0.jar ...
[2025-06-02T02:03:36.220+0000] {subprocess.py:93} INFO - 	[SUCCESSFUL ] org.apache.spark#spark-sql-kafka-0-10_2.12;3.2.0!spark-sql-kafka-0-10_2.12.jar (955ms)
[2025-06-02T02:03:36.353+0000] {subprocess.py:93} INFO - downloading https://repo1.maven.org/maven2/org/apache/spark/spark-token-provider-kafka-0-10_2.12/3.2.0/spark-token-provider-kafka-0-10_2.12-3.2.0.jar ...
[2025-06-02T02:03:36.556+0000] {subprocess.py:93} INFO - 	[SUCCESSFUL ] org.apache.spark#spark-token-provider-kafka-0-10_2.12;3.2.0!spark-token-provider-kafka-0-10_2.12.jar (334ms)
[2025-06-02T02:03:36.689+0000] {subprocess.py:93} INFO - downloading https://repo1.maven.org/maven2/org/apache/kafka/kafka-clients/2.8.0/kafka-clients-2.8.0.jar ...
[2025-06-02T02:03:44.725+0000] {subprocess.py:93} INFO - 	[SUCCESSFUL ] org.apache.kafka#kafka-clients;2.8.0!kafka-clients.jar (8169ms)
[2025-06-02T02:03:44.857+0000] {subprocess.py:93} INFO - downloading https://repo1.maven.org/maven2/com/google/code/findbugs/jsr305/3.0.0/jsr305-3.0.0.jar ...
[2025-06-02T02:03:45.060+0000] {subprocess.py:93} INFO - 	[SUCCESSFUL ] com.google.code.findbugs#jsr305;3.0.0!jsr305.jar (334ms)
[2025-06-02T02:03:45.192+0000] {subprocess.py:93} INFO - downloading https://repo1.maven.org/maven2/org/apache/commons/commons-pool2/2.6.2/commons-pool2-2.6.2.jar ...
[2025-06-02T02:03:45.667+0000] {subprocess.py:93} INFO - 	[SUCCESSFUL ] org.apache.commons#commons-pool2;2.6.2!commons-pool2.jar (605ms)
[2025-06-02T02:03:45.798+0000] {subprocess.py:93} INFO - downloading https://repo1.maven.org/maven2/org/spark-project/spark/unused/1.0.0/unused-1.0.0.jar ...
[2025-06-02T02:03:46.021+0000] {subprocess.py:93} INFO - 	[SUCCESSFUL ] org.spark-project.spark#unused;1.0.0!unused.jar (353ms)
[2025-06-02T02:03:46.154+0000] {subprocess.py:93} INFO - downloading https://repo1.maven.org/maven2/org/apache/hadoop/hadoop-client-runtime/3.3.1/hadoop-client-runtime-3.3.1.jar ...
[2025-06-02T02:05:00.858+0000] {subprocess.py:93} INFO - 	[SUCCESSFUL ] org.apache.hadoop#hadoop-client-runtime;3.3.1!hadoop-client-runtime.jar (74836ms)
[2025-06-02T02:05:00.991+0000] {subprocess.py:93} INFO - downloading https://repo1.maven.org/maven2/org/lz4/lz4-java/1.7.1/lz4-java-1.7.1.jar ...
[2025-06-02T02:05:02.697+0000] {subprocess.py:93} INFO - 	[SUCCESSFUL ] org.lz4#lz4-java;1.7.1!lz4-java.jar (1838ms)
[2025-06-02T02:05:02.830+0000] {subprocess.py:93} INFO - downloading https://repo1.maven.org/maven2/org/xerial/snappy/snappy-java/1.1.8.4/snappy-java-1.1.8.4.jar ...
[2025-06-02T02:05:07.957+0000] {subprocess.py:93} INFO - 	[SUCCESSFUL ] org.xerial.snappy#snappy-java;1.1.8.4!snappy-java.jar(bundle) (5259ms)
[2025-06-02T02:05:08.089+0000] {subprocess.py:93} INFO - downloading https://repo1.maven.org/maven2/org/slf4j/slf4j-api/1.7.30/slf4j-api-1.7.30.jar ...
[2025-06-02T02:05:08.384+0000] {subprocess.py:93} INFO - 	[SUCCESSFUL ] org.slf4j#slf4j-api;1.7.30!slf4j-api.jar (425ms)
[2025-06-02T02:05:08.517+0000] {subprocess.py:93} INFO - downloading https://repo1.maven.org/maven2/org/apache/hadoop/hadoop-client-api/3.3.1/hadoop-client-api-3.3.1.jar ...
[2025-06-02T02:05:37.716+0000] {subprocess.py:93} INFO - 	[SUCCESSFUL ] org.apache.hadoop#hadoop-client-api;3.3.1!hadoop-client-api.jar (29332ms)
[2025-06-02T02:05:37.855+0000] {subprocess.py:93} INFO - downloading https://repo1.maven.org/maven2/org/apache/htrace/htrace-core4/4.1.0-incubating/htrace-core4-4.1.0-incubating.jar ...
[2025-06-02T02:05:38.774+0000] {subprocess.py:93} INFO - 	[SUCCESSFUL ] org.apache.htrace#htrace-core4;4.1.0-incubating!htrace-core4.jar (1056ms)
[2025-06-02T02:05:38.906+0000] {subprocess.py:93} INFO - downloading https://repo1.maven.org/maven2/commons-logging/commons-logging/1.1.3/commons-logging-1.1.3.jar ...
[2025-06-02T02:05:39.064+0000] {subprocess.py:93} INFO - 	[SUCCESSFUL ] commons-logging#commons-logging;1.1.3!commons-logging.jar (290ms)
[2025-06-02T02:05:39.065+0000] {subprocess.py:93} INFO - :: resolution report :: resolve 19972ms :: artifacts dl 123800ms
[2025-06-02T02:05:39.066+0000] {subprocess.py:93} INFO - 	:: modules in use:
[2025-06-02T02:05:39.066+0000] {subprocess.py:93} INFO - 	com.google.code.findbugs#jsr305;3.0.0 from central in [default]
[2025-06-02T02:05:39.066+0000] {subprocess.py:93} INFO - 	commons-logging#commons-logging;1.1.3 from central in [default]
[2025-06-02T02:05:39.067+0000] {subprocess.py:93} INFO - 	org.apache.commons#commons-pool2;2.6.2 from central in [default]
[2025-06-02T02:05:39.067+0000] {subprocess.py:93} INFO - 	org.apache.hadoop#hadoop-client-api;3.3.1 from central in [default]
[2025-06-02T02:05:39.067+0000] {subprocess.py:93} INFO - 	org.apache.hadoop#hadoop-client-runtime;3.3.1 from central in [default]
[2025-06-02T02:05:39.068+0000] {subprocess.py:93} INFO - 	org.apache.htrace#htrace-core4;4.1.0-incubating from central in [default]
[2025-06-02T02:05:39.074+0000] {subprocess.py:93} INFO - 	org.apache.kafka#kafka-clients;2.8.0 from central in [default]
[2025-06-02T02:05:39.075+0000] {subprocess.py:93} INFO - 	org.apache.spark#spark-sql-kafka-0-10_2.12;3.2.0 from central in [default]
[2025-06-02T02:05:39.076+0000] {subprocess.py:93} INFO - 	org.apache.spark#spark-token-provider-kafka-0-10_2.12;3.2.0 from central in [default]
[2025-06-02T02:05:39.077+0000] {subprocess.py:93} INFO - 	org.lz4#lz4-java;1.7.1 from central in [default]
[2025-06-02T02:05:39.077+0000] {subprocess.py:93} INFO - 	org.slf4j#slf4j-api;1.7.30 from central in [default]
[2025-06-02T02:05:39.079+0000] {subprocess.py:93} INFO - 	org.spark-project.spark#unused;1.0.0 from central in [default]
[2025-06-02T02:05:39.079+0000] {subprocess.py:93} INFO - 	org.xerial.snappy#snappy-java;1.1.8.4 from central in [default]
[2025-06-02T02:05:39.080+0000] {subprocess.py:93} INFO - 	---------------------------------------------------------------------
[2025-06-02T02:05:39.080+0000] {subprocess.py:93} INFO - 	|                  |            modules            ||   artifacts   |
[2025-06-02T02:05:39.081+0000] {subprocess.py:93} INFO - 	|       conf       | number| search|dwnlded|evicted|| number|dwnlded|
[2025-06-02T02:05:39.082+0000] {subprocess.py:93} INFO - 	---------------------------------------------------------------------
[2025-06-02T02:05:39.082+0000] {subprocess.py:93} INFO - 	|      default     |   13  |   13  |   13  |   0   ||   13  |   13  |
[2025-06-02T02:05:39.083+0000] {subprocess.py:93} INFO - 	---------------------------------------------------------------------
[2025-06-02T02:05:39.083+0000] {subprocess.py:93} INFO - :: retrieving :: org.apache.spark#spark-submit-parent-ed84d9e7-fd06-49e3-bc87-fc4719c53d8a
[2025-06-02T02:05:39.084+0000] {subprocess.py:93} INFO - 	confs: [default]
[2025-06-02T02:05:56.273+0000] {subprocess.py:93} INFO - 	13 artifacts copied, 0 already retrieved (59181kB/17201ms)
[2025-06-02T02:06:16.158+0000] {job.py:229} INFO - Heartbeat recovered after 13.30 seconds
[2025-06-02T02:06:18.024+0000] {subprocess.py:93} INFO - 25/06/02 02:06:18 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2025-06-02T02:06:24.590+0000] {subprocess.py:93} INFO - python3: can't open file '/app/log_consumer.py': [Errno 2] No such file or directory
[2025-06-02T02:06:24.636+0000] {subprocess.py:93} INFO - 25/06/02 02:06:24 INFO ShutdownHookManager: Shutdown hook called
[2025-06-02T02:06:24.636+0000] {subprocess.py:93} INFO - 25/06/02 02:06:24 INFO ShutdownHookManager: Deleting directory /tmp/spark-15ab5cb4-47f6-482b-ac71-6c33d069fac0
[2025-06-02T02:06:25.996+0000] {subprocess.py:97} INFO - Command exited with return code 2
[2025-06-02T02:06:26.067+0000] {taskinstance.py:3310} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/bash.py", line 249, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 2.
[2025-06-02T02:06:26.073+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=log_monitoring_dag, task_id=start_spark_consumer, run_id=manual__2025-06-02T01:38:51.955055+00:00, execution_date=20250602T013851, start_date=20250602T020309, end_date=20250602T020626
[2025-06-02T02:06:26.157+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-06-02T02:06:26.158+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 76 for task start_spark_consumer (Bash command failed. The command returned a non-zero exit code 2.; 860)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 115, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3004, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3158, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3182, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/bash.py", line 249, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 2.
[2025-06-02T02:06:26.171+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-06-02T02:06:26.193+0000] {taskinstance.py:3900} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-06-02T02:06:26.248+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
